src/resolve/resolved-dns-query.c

```diff
@@ -38,19 +38,16 @@ static int dns_query_candidate_new(DnsQueryCandidate **ret, DnsQuery *q, DnsScop
 }
 
 static void dns_query_candidate_stop(DnsQueryCandidate *c) {
-        DnsTransaction *t;
-
         assert(c);
 
-        while ((t = set_steal_first(c->transactions))) {
+        for (DnsTransaction *t; (t = set_steal_first(c->transactions)); ) {
```

Comment:
that doesn't look like an improvement. typically the second expression in the for () is side-effect free. It's really weird to misuse a for() loop as a while() loop here...

Reply:
I'll drop that chunk.